src/Control/Effect/Fail.hs

```diff
@@ -18,11 +18,8 @@ import Data.Coerce
 import Prelude hiding (fail)
 
 newtype Fail (m :: * -> *) k = Fail String
-  deriving (Functor)
-
-instance HFunctor Fail where
-  hmap _ = coerce
-  {-# INLINE hmap #-}
```

Comment:
Do we lose inlining with the default definition? Or does the `INLINE` pragma on the default definition apply to each derived instance?

Reply:
The INLINE pragma applies to each derived instance 🎉